tests/app/data/test_schema_id_regex.py

```diff
@@ -0,0 +1,133 @@
+        self.errors = []
+        self.schema = json.load(schema_file)
+
+    def test_default_id_should_pass_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id()
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 0)
+
+    def test_guid_should_pass_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('14ba4707-321d-441d-8d21-b8367366e766')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 0)
+
+    def test_id_with_hyphenated_names_should_pass_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('name-with-hyphens')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 0)
+
+    def test_id_with_numeric_should_pass_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('this-is-a-valid-id-0')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 0)
+
+    def test_id_with_numeric_at_start_should_pass_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('0-this-is-a-valid-id-0')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 0)
+
+    def test_id_with_punctuation_should_fail_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('!n0t-@-valid-id')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 1)
+
+    def test_id_with_spaces_should_fail_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('not a valid id')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 1)
+
+    def test_id_with_capital_letters_should_fail_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('NOT-A-VALID-ID')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 1)
+
+    def test_id_with_underscores_should_pass_validation(self):
+        # Given
+        json_to_validate = create_schema_with_id('not_a_valid_id')
+
+        # When
+        errors = self.validate_json_against_schema(json_to_validate, self.schema)
+
+        # Then
+        self.assertEqual(len(errors), 1)
+
+    @staticmethod
+    def validate_json_against_schema(json_to_validate, schema):
```

Comment:
Codecov seems to be complaining that this doesn't appear with test_ in the name. Given it's static, is it worth moving it outside the class as with create_schema_with_id?